### ISSUE TITLE
Ohara 0.9.0 product testing report

### ISSUE THREAD

**Issue by saivirtue** [oharastream/ohara#3706]
## 測試項目

<details>
<summary><b>1. 基本測試</b></summary>

- [ ] 確認畫面（UI）與設計稿（Design）是否一致
  - 顏色、字型、符號、圖片、...
- [ ] 檢查所有畫面的 Title 名稱是否正確
- [ ] 所有連結是否有前往正確頁面
- [ ] 切換視窗大小時，有無跑版情形
- [ ] 切換視窗大小後，文字與圖形是否竄行
- 表單
  - [ ] 任何一個必填欄位未填寫，不能送出
  - [ ] 表單送出中，應該明確表達正在處理中的訊號
  - [ ] 表單送出中，應該要凍結表單（除非有提供取消功能）
  - [ ] 任何表單至少要有 CTA 及 Cancel 兩類按紐 (CTA: Call-To-Action)
- 訊息
  - [ ] 成功訊息，應該要明確指出完成對象
  - [ ] 錯誤訊息，應該要明確指出錯誤對象
  - [ ] 確認訊息
    - [ ] 刪除確認，應該要明確指出欲刪除對象
- 按鈕
  - [ ] 任何一個刪除按鈕，都要有刪除確認提示
- 鼠標
  - [ ] 移動到所有可以點擊的元素，要出現 pointer cursor
      - Button, Link icon, Link text, ...
</details>

<details>
<summary><b>2. 功能測試: <a href="https://github.com/oharastream/ohara/blob/master/docs/testcase/pipeline-jdbc-topic-stream-topic-hdfs.md">JDBC -> Stream -> HDFS</a></b></summary>

- [x] Create a new workspace with three nodes
- [ ] Create two topic and upload a stream app jar into the workspace
- [x] Create a new pipeline and add some components into pipeline
- [x] 連接 JDBC Source -> Topic 1 -> Stream App -> Topic 2 -> HDFS Sink
- [x] 連接 Perf Source -> Topic 
- [x] Prepare required folders and test data on the JDBC server
- [x] Start all connectors and stream app
- [x] Verify which test data was successfully dumped to HDFS
</details>

<details>
<summary><b>3. 安裝測試:<a href="https://github.com/oharastream/ohara/wiki/Test-case:-Install-case">Install case</a></b></summary>

- [x] Confirm your use the hardware specification
- [x] Confirm your use version
- [x] Install the Kubernetes
- [x] Running the Ohara Configuration service
- [x] Running the Ohara Manager service

</details>
### 測試環境：
- Ohara
  - Mode: k8s 
  - Version: 0.9.0-SNAPSHOT
  - Build date: 2019-08-17 17:09:26
- 測試工具
  - 瀏覽器：Chrome  76.0.3809.100 
  - 最小螢幕寬度：1200

### 基本測試：

- [Manager] Cannot display the connectors, streams in plugin card (#3747)
- [Manager] Set metrics display setting to off when a user agent visit our App for the first time (#3732)
- Early avoid duplicated workspace name when creating a new one (#3746)
- [Manager] Add default workspace name in quick mode (#3752)
- [Manager] Remove the link from UI's Ohara logo (#3777)
- [Manager] Add a hint on the "into the worksapce" button (#3786)
- [Manager] Remove the settings button from topic cells (#3780)

### 功能測試：

- BasicTestPerformance4Ftp#removeFtpFolder 不能移除空資料夾 (#3707)
- [Manager] Unable to get the correct group when changing workspace (#3705)
- [Manager] Delete a pipeline shouldn't delete the public topics used in it (#3733)
- Before start the connector, await worker http server completed for performance test (#3665)
- Change the permission of connector.class from EDITABLE to CREATE_ONLY (#3745)
- Configurator log should be able to fetch even if no workspace (#3768)
- [Manager] Prevent users from adding more topic into the paper while a new topic is adding (#3766)
- the run-time information of creating node is incorrect - it is always unavailable and resources are empty (#3785)
- Hostname should not be editable in docker mode (#3784)
- [Manager] Stream names are not showing up in Toolbox after uploaded (#3754)

**Comment by saivirtue** [comment]
@oharastream/developer 
If you have found any issue or problems about version `0.9`, please add the issue number here.

**Comment by wu87988622** [comment]
Graph updates automatically when connector updates #3693

**Comment by jackyoh** [comment]
- [ ] BasicTestPerformance4Ftp#removeFtpFolder 不能移除空資料夾 #3707

**Comment by eechih** [comment]
- [ ] [Manager] Cannot change ApiContext's group when change workspace (#3705) @eechih

**Comment by konekoya** [comment]
- [ ] [Manager] Delete a pipeline shouldn't delete the public topics used in it

**Comment by chia7712** [comment]
@oharastream/frontend 我在docker(ssh) mode裡面加入節點後沒看到資源顯示，但我透過APIs去敲是可以看到的

![註解 2020-01-08 102251](https://user-images.githubusercontent.com/6234750/71945217-342de300-3201-11ea-9480-efdfaee0f2c0.jpg)

![註解 2020-01-08 1022512](https://user-images.githubusercontent.com/6234750/71945278-532c7500-3201-11ea-8507-69bae8fc03ae.jpg)

**Comment by chia7712** [comment]
加入節點後，回到建立workspace的頁面卻沒有任何節點顯示

![1](https://user-images.githubusercontent.com/6234750/71945337-853dd700-3201-11ea-8ad2-a809cef26902.jpg)

![2](https://user-images.githubusercontent.com/6234750/71945371-9d155b00-3201-11ea-953f-43081f5e2e26.jpg)

**Comment by chia7712** [comment]
加入plugins後沒有顯示內容(例如有哪些connectors, streams)

![3](https://user-images.githubusercontent.com/6234750/71945680-6e4bb480-3202-11ea-9739-b2e57e1ab742.jpg)

**Comment by chia7712** [comment]
connector的線已經連結到一個topic，但在connector的設定裡面卻依然沒有顯示已經選擇的topic

![1](https://user-images.githubusercontent.com/6234750/71946504-08acf780-3205-11ea-95f7-9042d96ee1ea.jpg)

![2](https://user-images.githubusercontent.com/6234750/71946505-0b0f5180-3205-11ea-8f48-e7e7b2a058c4.jpg)

**Comment by saivirtue** [comment]
> connector的線已經連結到一個topic，但在connector的設定裡面卻依然沒有顯示已經選擇的topic
> 
> ![1](https://user-images.githubusercontent.com/6234750/71946504-08acf780-3205-11ea-95f7-9042d96ee1ea.jpg)
> 
> ![2](https://user-images.githubusercontent.com/6234750/71946505-0b0f5180-3205-11ea-8f48-e7e7b2a058c4.jpg)

It seems duplicated to #3693? /cc @wu87988622

**Comment by chia7712** [comment]
duration型態的預設值不見了

![3](https://user-images.githubusercontent.com/6234750/71946881-0008f100-3206-11ea-8fe9-965694b7a597.jpg)

**Comment by chia7712** [comment]
- [ ] Change the permission of connector.class from EDITABLE to CREATE_ONLY (#3745)

**Comment by saivirtue** [comment]
- [ ] Early avoid duplicated workspace name when creating a new one #3746

**Comment by eechih** [comment]
> 加入plugins後沒有顯示內容(例如有哪些connectors, streams)

- [ ] [Manager] Cannot display the connectors, streams in plugin card (#3747 )

**Comment by konekoya** [comment]
[ ] [Manager] Stream names are not showing up in Toolbox after uploaded

**Comment by chia7712** [comment]
fail to observe configurator log from UI (but API works well)

![註解 2020-01-08 144701](https://user-images.githubusercontent.com/6234750/71956407-0dce6e80-3226-11ea-95e0-c454eaf3d2d4.jpg)

乍看之下很像是UI嘗試從"已經加入的節點"找configurator log(?)，可是configurator可能根本不在已經加入的節點列表裡面。或許UI應該直接打${configurator_url}/logs/configurator就好，而不用去在意原本的nodes列表

**Comment by saivirtue** [comment]
> fail to observe configurator log from UI (but API works well)
> 乍看之下很像是UI嘗試從"已經加入的節點"找configurator log(?)，可是configurator可能根本不在已經加入的節點列表裡面。或許UI應該直接打${configurator_url}/logs/configurator就好，而不用去在意原本的nodes列表

I tested the same behavior in our [Demo env](http://ohara-demo-00:5050/), and it works well:
![image](https://user-images.githubusercontent.com/8088616/71957064-b0d3b800-3227-11ea-8b28-e8e4938a4357.png)

**Comment by chia7712** [comment]
@saivirtue 你嘗試看看讓configurator不是位於nodes列表 (如我上面的敘述)

例如node00跑configurator，然後只把node01加入configurator，並把其他服務啟動在node01

**Comment by saivirtue** [comment]
> @saivirtue 你嘗試看看讓configurator不是位於nodes列表 (如我上面的敘述)
> 
> 例如node00跑configurator，然後只把node01加入configurator，並把其他服務啟動在node01

It seems like something wrong in **docker mode**
I will update the test result later...

**Comment by saivirtue** [comment]
> @saivirtue 你嘗試看看讓configurator不是位於nodes列表 (如我上面的敘述)
> 
> 例如node00跑configurator，然後只把node01加入configurator，並把其他服務啟動在node01

The reason UI displays "nothing" is because of API return "empty log".
The default `sinceSeconds` in UI is using 600, but your native API is using empty (which means all data)

BTW, we use the node name from logs.hostname of API response, node list doesn't participate in it :)

**Comment by saivirtue** [comment]
Configurator log should be able to fetch even if no workspace
![image](https://user-images.githubusercontent.com/8088616/71959209-1eceae00-322d-11ea-81f5-3e7932535197.png)

- [ ] Configurator log should be able to fetch even if no workspace (#3768)

**Comment by chia7712** [comment]
> The default sinceSeconds in UI is using 600, but your native API is using empty (which means all data)

看起來應該是APIs的錯誤 (see https://github.com/oharastream/ohara/issues/3770)

**Comment by chia7712** [comment]
> Configurator log should be able to fetch even if no workspace

這個我剛剛也有遇到，一定要先建立才可以看logs :(

**Comment by chia7712** [comment]
topic (pipeline only)的設定按鈕按了沒有反應 (當時沒有任何運作中的connectors, streams)

![1](https://user-images.githubusercontent.com/6234750/71961478-5855e800-3232-11ea-916f-a360ae666f44.jpg)

**Comment by chia7712** [comment]
pipeline上的connector已經啟動了(connector and pipeline APIs都顯示running)，但是畫面依然顯示stopped

ps.  已經重複刷新頁面，依然是stopped

![2](https://user-images.githubusercontent.com/6234750/71962367-3a898280-3234-11ea-97e7-043c9e7c89d7.jpg)
![0](https://user-images.githubusercontent.com/6234750/71962368-3a898280-3234-11ea-9aa0-0b7e9d7764c0.jpg)
![1](https://user-images.githubusercontent.com/6234750/71962369-3b221900-3234-11ea-895d-2ac4a88db6ce.jpg)

**Comment by konekoya** [comment]
- [ ] [Manager] Remove the link from UI's Ohara logo

**Comment by chia7712** [comment]
首頁的文字typo:

Ohara is a scalable streaming platform that allows users to easily organize**d** their input, output, and streaming applications with a clean and comprehensive GUI

**Comment by chia7712** [comment]
"into the worksapce" 無法點擊

![註解 2020-01-09 005211](https://user-images.githubusercontent.com/6234750/71998284-4a2bba00-327a-11ea-8253-18ee1ca7b97a.jpg)

**Comment by chia7712** [comment]
duration型態的資料存放有誤，應該是"1 seconds"而不是"1"

![2](https://user-images.githubusercontent.com/6234750/71998827-4482a400-327b-11ea-9681-7498e4b6af38.jpg)
![註解 2020-01-09 005211](https://user-images.githubusercontent.com/6234750/71998828-451b3a80-327b-11ea-9886-0b3485041fa7.jpg)

**Comment by konekoya** [comment]
> 首頁的文字typo:

Not that good at grammar, but I believe verbs after the "to" should be "Organize" not "Organized" :)

**Comment by konekoya** [comment]
> "into the worksapce" 無法點擊

It's intended. Because it's the current workspace

**Comment by konekoya** [comment]
> topic (pipeline only)的設定按鈕按了沒有反應 (當時沒有任何運作中的connectors, streams)

Hmm, the config button should be removed from topics

**Comment by konekoya** [comment]
-[ ] [Manager] Remove the settings button from topic cells #3780

**Comment by konekoya** [comment]
- [ ] [Manager] Remove the settings button from topic cells (#3780)

**Comment by saivirtue** [comment]
> 首頁的文字typo:
> 
> Ohara is a scalable streaming platform that allows users to easily organize**d** their input, output, and streaming applications with a clean and comprehensive GUI

The verb should be infinitives here since it's "allows...to + V"

**Comment by saivirtue** [comment]
> "into the worksapce" 無法點擊
> 
> ![註解 2020-01-09 005211](https://user-images.githubusercontent.com/6234750/71998284-4a2bba00-327a-11ea-8253-18ee1ca7b97a.jpg)

If there is one workspace only, the "into the workspace" means current workspace (which should be disabled here).

**Comment by konekoya** [comment]
> If there is one workspace only, the "into the workspace" means current workspace (which should be disabled here).

Maybe we can provide a tooltip as a hint like: "This is the workspace that you're at now"?

**Comment by saivirtue** [comment]
> > If there is one workspace only, the "into the workspace" means current workspace (which should be disabled here).
> 
> Maybe we can provide a tooltip as a hint like: "This is the workspace that you're at now"?

+1

**Comment by konekoya** [comment]
> Maybe we can provide a tooltip as a hint like: "This is the workspace that you're at now"?

/cc @eechih @wu87988622 for feedback on this?

**Comment by saivirtue** [comment]
node 編輯頁面不應該可以修改hostname (會產生新的node)
![image](https://user-images.githubusercontent.com/8088616/72034935-61e25d00-32d1-11ea-9ddf-1da98a30ebef.png)

**Comment by eechih** [comment]
> Maybe we can provide a tooltip as a hint like: "This is the workspace that you're at now"?

+1

**Comment by chia7712** [comment]
> 我在docker(ssh) mode裡面加入節點後沒看到資源顯示，但我透過APIs去敲是可以看到的

 這是APIs的問題 (#3785)

**Comment by konekoya** [comment]
- [ ] [Manager] Handle same name connector error #3797

**Comment by saivirtue** [comment]
![image](https://user-images.githubusercontent.com/8088616/72050685-b222e480-32fc-11ea-9397-c6ea16a113ec.png)

- [ ] [Manager] Could not save settings of connector if absent topic field #3802

**Comment by saivirtue** [comment]
![image](https://user-images.githubusercontent.com/8088616/72051000-61f85200-32fd-11ea-9831-341f81e58036.png)

- [ ] [Manager] Could not edit stream settings (#3803)

**Comment by chia7712** [comment]
![註解 2020-02-20 113647](https://user-images.githubusercontent.com/6234750/74898601-4b054f00-53d5-11ea-9653-48fcf26d7c9b.jpg)

從pipeline-only topic無法拉線連接到JsonOut

**Comment by chia7712** [comment]
![註解 2020-02-20 113647](https://user-images.githubusercontent.com/6234750/74899022-a7b53980-53d6-11ea-8f58-662bbcb3bd3d.jpg)

Type.Duration的預設值沒有顯示

**Comment by vitojeng** [comment]
Quickstart VM --> Nodes --> ADD NODE --> 填入正確資料 --> ADD
* Dialog 直接關閉, 畫面沒有任何改變.
* 過了數秒後, 左下方會出現: Failed to create node xxx.xxx.xxx.xxx
* Event log 資訊顯示如下:
![screenshot 18](https://user-images.githubusercontent.com/475186/75002728-570a1300-54a0-11ea-8ad1-a56573beca80.jpg)

**Comment by vitojeng** [comment]
QuickstartVM(Mem 12GB), 操作步驟:
* Add node, Create workspace 成功
* 建立 pipeline, 建立 ftp source connector, ftp sink connector, 建立 source/sink 連線
* 在 ftp source connector 填入 settings, 然後按下 **Start**
* ftp source connector 顯示橘燈(Pending), 然後一直在橘燈狀態

**Browser tools 觀察, API 出現錯誤**
Request URL: http://192.168.56.108:5050/api/connectors/ftpsource1?group=fcfb6f54c
Error:
```
"oharastream.ohara.common.exception.OharaFileSystemException: Could not parse response code.\nServer Reply: SSH-2.0-OpenSSH_7.6p1 Ubuntu-4ubuntu0.3\n\tat oharastream.ohara.client.filesystem.FileSystem.wrap(FileSystem.scala:102)\n\tat oharastream.ohara.client.filesystem.FileSystem.wrap$(FileSystem.scala:98)\n\tat oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.wrap(FtpFileSystem.scala:135)\n\tat oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.exists(FtpFileSystem.scala:142)\n\tat oharastream.ohara.client.filesystem.FileSystem.nonExists(FileSystem.scala:51)\n\tat oharastream.ohara.client.filesystem.FileSystem.nonExists$(FileSystem.scala:51)\n\tat oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.nonExists(FtpFileSystem.scala:135)\n\tat oharastream.ohara.connector.ftp.FtpSource._start(FtpSource.scala:48)\n\tat oharastream.ohara.kafka.connector.RowSourceConnector.start(RowSourceConnector.java:120)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:110)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:135)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.transitionTo(WorkerConnector.java:195)\n\tat org.apache.kafka.connect.runtime.Worker.startConnector(Worker.java:257)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder.startConnector(DistributedHerder.java:1183)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder.access$1400(DistributedHerder.java:125)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder$15.call(DistributedHerder.java:1199)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder$15.call(DistributedHerder.java:1195)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: org.apache.commons.net.MalformedServerReplyException: Could not parse response code.\nServer Reply: SSH-2.0-OpenSSH_7.6p1 Ubuntu-4ubuntu0.3\n\tat org.apache.commons.net.ftp.FTP.__getReply(FTP.java:344)\n\tat org.apache.commons.net.ftp.FTP.__getReply(FTP.java:300)\n\tat org.apache.commons.net.ftp.FTP._connectAction_(FTP.java:438)\n\tat org.apache.commons.net.ftp.FTPClient._connectAction_(FTPClient.java:962)\n\tat org.apache.commons.net.ftp.FTPClient._connectAction_(FTPClient.java:950)\n\tat org.apache.commons.net.SocketClient._connect(SocketClient.java:244)\n\tat org.apache.commons.net.SocketClient.connect(SocketClient.java:202)\n\tat oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$FtpClientImpl.connectIfNeeded(FtpClient.scala:318)\n\tat oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$FtpClientImpl.exist(FtpClient.scala:476)\n\tat oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.$anonfun$exist$1(FtpClient.scala:303)\n\tat scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)\n\tat oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.retry(FtpClient.scala:281)\n\tat oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.exist(FtpClient.scala:303)\n\tat oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.$anonfun$exists$1(FtpFileSystem.scala:143)\n\tat scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)\n\tat oharastream.ohara.client.filesystem.FileSystem.wrap(FileSystem.scala:100)\n\t... 20 more\n"
```

**Docker logs 觀察, Worker 載入 connector 出現錯誤**
![screenshot 19](https://user-images.githubusercontent.com/475186/75011329-541c1c00-54ba-11ea-8298-55dbabffc602.jpg)

**Comment by konekoya** [comment]
Display name is not capitalized 

![Display name is not capitalized image](https://www.dropbox.com/s/iyza40olqp9ihkz/Screen%20Shot%202020-02-24%20at%209.35.24%20AM.png?raw=1)

**Comment by eechih** [comment]
We should confirm before deleting

<img width="1212" alt="Screen Shot 2020-02-24 at 9 52 43 AM" src="https://user-images.githubusercontent.com/703126/75124723-04249b80-56ec-11ea-8133-542454cb1c99.png">

**Comment by vitojeng** [comment]
Error "Failed to inspect topic XX Info" when open new window(#4109)

**Comment by chia7712** [comment]
![註解 2020-02-24 130628](https://user-images.githubusercontent.com/6234750/75129915-8326cd80-5706-11ea-9388-e76c93b8169c.jpg)

重開所有服務後，event logs裡面依舊有前世的記憶

**Comment by chia7712** [comment]
![註解 2020-02-24 130628](https://user-images.githubusercontent.com/6234750/75130139-b322a080-5707-11ea-9a30-a8593836a05d.jpg)

錯誤訊息不正確

**Comment by chia7712** [comment]
![註解 2020-02-24 130628](https://user-images.githubusercontent.com/6234750/75130168-d8171380-5707-11ea-92b3-1c77d4a807a6.jpg)

不存在的key但允許儲存，儲存完後再開會變成另外一個key

![2](https://user-images.githubusercontent.com/6234750/75130194-f715a580-5707-11ea-8f60-56cf5b400ef8.jpg)

重現方式:

1. 先選任意一個可用的key
1. 手動更動該key
1. 點選儲存

**Comment by chia7712** [comment]
![註解 2020-02-24 130628](https://user-images.githubusercontent.com/6234750/75130251-4065f500-5708-11ea-8a1a-b05fbc9b9e27.jpg)

pipelinename應該換成拼法正確的字，或者加上編號的單字 (如pipeline0)

**Comment by chia7712** [comment]
點選"delete this pipeline"後並沒有幫忙刪除該pipeline上原有的元件

重現方式:

1. 建立PerfSource
1. 建立pipeline-only topic
1. 啟動PerfSource
1. 點選"delete this pipeline"

**Comment by vitojeng** [comment]
在 connector 因後端出現錯誤而進入 FAILED 狀態下, 若使用者需要修改 property, 此時 connector 需要先 stop, 才能 edit.
因此, 在 FAILED 時應該要 disable "Edit" button.

**Comment by vitojeng** [comment]
Develop Tools -> LOGS -> select worker
* 顯示 logs 內容
* Scrollbar 捲至最後
* 內容慢慢自上移動(接 Up or 使用滑鼠), 重覆幾次後就會發現捲到最上面了

**Comment by vitojeng** [comment]
有時候會出現下列情形(目前尚無法重現):
在 Develop Tools, 從  LOGS tab 切換至 TOPICS tab 後即跳至 500 page:
> 500: Oops, something went terribly wrong!
> You either tried some shady route or you came here by mistake. Whichever it is, try using the navigation

refresh page 後似乎沒遇到什麼問題~

**Comment by saivirtue** [comment]
> Log 內容自動跳到最前面
> 操作順序: Develop Tools -> LOGS -> select worker
> 
> * 顯示 logs 內容
> * Scrollbar 捲至最後
> * 內容慢慢向上移動(按 Up key or 使用滑鼠), 重覆幾次後就會發現自動跳到最上面了

This may be related to #4091, could you pull the PR and test the same scenario again?

**Comment by saivirtue** [comment]
![image](https://user-images.githubusercontent.com/8088616/75211145-0e16cf00-57be-11ea-975c-52ef79993a81.png)

The `error.folder` setting in FtpSource is defined with Optional, but the backend require it...

**Comment by vitojeng** [comment]
Failed to inspect topic XX info after delete topic(#4138)

**Comment by eechih** [comment]
FTP Sink 的 File Need Header 無法打勾

![Screen Shot 2020-02-25 at 3 16 12 PM](https://user-images.githubusercontent.com/703126/75223771-213b9600-57e2-11ea-9b3a-eb755ad87ec4.png)

**Comment by eechih** [comment]
> ![註解 2020-02-24 130628](https://user-images.githubusercontent.com/6234750/75129915-8326cd80-5706-11ea-9388-e76c93b8169c.jpg)
> 
> 重開所有服務後，event logs裡面依舊有前世的記憶

@chia7712  我會在下個版本處理 #4143

**Comment by vitojeng** [comment]
Minor: a vertical line in the TOPICS tab.
![screenshot 33](https://user-images.githubusercontent.com/475186/75226114-bfc9f600-57e6-11ea-8aa6-b1e00d766197.jpg)

**Comment by konekoya** [comment]
> 從pipeline-only topic無法拉線連接到JsonOut

@chia7712 Are you still experiencing this? I'm unable to repro this on my machine...

**Comment by chia7712** [comment]
> Are you still experiencing this? I'm unable to repro this on my machine...

I fail to produce this error :(

**Comment by eechih** [comment]
**Failed to create shared topic**

**Describe the bug**
建立 Shared Topic 遇到啟動失敗後，沒有出現在 Toolbox 或 Topics，也無法再次建立（使用相同名稱）。

**To Reproduce**
1. 新增 Shared Topic (Name: sharedtopic11, Partitions: 3, Replication factor: 3)
2. 錯誤訊息 "Failed to start topic sharedtopic11"
3. 再次新增 Shared Topic (Name: sharedtopic11, Partitions: 1, Replication factor: 1) 
4. 錯誤訊息 "Failed to create topic sharedtopic11"

<img width="962" alt="Screen Shot 2020-02-26 at 1 46 24 PM" src="https://user-images.githubusercontent.com/703126/75315945-0c710800-589f-11ea-9a0a-dd4d9fb30eca.png">

<img width="959" alt="Screen Shot 2020-02-26 at 1 46 56 PM" src="https://user-images.githubusercontent.com/703126/75315965-14c94300-589f-11ea-977c-ba57c82dfb06.png">

**Comment by eechih** [comment]
**沒有加入到 Pipeline 的 Topic，不應該出現在選單中**

<img width="958" alt="Screen Shot 2020-02-26 at 3 06 50 PM" src="https://user-images.githubusercontent.com/703126/75320669-27954500-58aa-11ea-8918-5fb7cbd966a8.png">

**Comment by eechih** [comment]
調整 number of task 後，存檔會失敗

<img width="956" alt="Screen Shot 2020-02-26 at 3 26 16 PM" src="https://user-images.githubusercontent.com/703126/75322056-0aae4100-58ad-11ea-810f-746cafd06c2a.png">
<img width="950" alt="Screen Shot 2020-02-26 at 3 26 33 PM" src="https://user-images.githubusercontent.com/703126/75322080-16016c80-58ad-11ea-9b45-4d75df60f0e6.png">

**Comment by konekoya** [comment]
> I fail to produce this error :(

Not problem. I will mark the issue as solve then :)

**Comment by eechih** [comment]
啟動中的 Stream，如果 Save Change 會造成 Stream 的狀態變成 Failed

<img width="1373" alt="Screen Shot 2020-02-26 at 4 41 23 PM" src="https://user-images.githubusercontent.com/703126/75327206-ef483380-58b6-11ea-8acf-b54b6b729546.png">

**Comment by vitojeng** [comment]
Minor: Event log 訊息太長時, row 高度會暴增, 建議將訊息截斷
![screenshot 40](https://user-images.githubusercontent.com/475186/75333365-fecc7a00-58c0-11ea-8e6f-e93ce92a640f.jpg)

**Comment by chia7712** [comment]
![註解 2020-03-03 111536](https://user-images.githubusercontent.com/6234750/75739624-5a26be00-5d40-11ea-9c8a-6d57ed5518df.jpg)

port的顯示不用逗號，那個不是薪水

**Comment by chia7712** [comment]
![註解 2020-03-03 111536](https://user-images.githubusercontent.com/6234750/75739712-9fe38680-5d40-11ea-82db-b8564ec3e565.jpg)

已經啟動的connector依然可以再次啟動?

**Comment by chia7712** [comment]
![註解 2020-03-03 111536](https://user-images.githubusercontent.com/6234750/75739773-c1dd0900-5d40-11ea-91d4-27cf92547988.jpg)

sink connector的nodes和metrics沒有顯示出來

**Comment by konekoya** [comment]
> sink connector的nodes和metrics沒有顯示出來

A fix is on the way... #4196

**Comment by konekoya** [comment]
> 已經啟動的connector依然可以再次啟動?

@eechih I think we have missed this... So I will file another issue to fix this 😅

**Comment by eechih** [comment]
> 已經啟動的connector依然可以再次啟動?

> @eechih I think we have missed this... So I will file another issue to fix this 😅

Ok~ Thanks @konekoya

**Comment by eechih** [comment]
![Screen Shot 2020-03-04 at 11 30 29 AM](https://user-images.githubusercontent.com/703126/75842681-f5d13080-5e0b-11ea-822e-e4b89458dae0.png)

**Comment by vitojeng** [comment]
Pipelines 清單自動只剩一個. refresh page 一開始有三個 pipeline:
![screenshot 48](https://user-images.githubusercontent.com/475186/75853305-c54bbf80-5e28-11ea-963b-51e85cd8e775.jpg)

靜置約 10 秒剩一個:
![screenshot 49](https://user-images.githubusercontent.com/475186/75853338-d5fc3580-5e28-11ea-8775-48b3fe8f5ab2.jpg)

**Comment by konekoya** [comment]
> 重現: 建一個 workspace, 接著再建三個 pipelines, 過一會兒應該就可發現

@wu87988622 it's happing again... 🙇

**Comment by vitojeng** [comment]
FtpSourceConnector 相關....重現: 
1. 填入相關 setting, 但是 input.folder 的資料夾填錯(不存在), 並連結 topic
2. Start connector

情形:
1. Response Error:
```
error: java.util.NoSuchElementException: /vito/source/files doesn't exist
	at oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$FtpClientImpl.fileType(FtpClient.scala:497)
	at oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.$anonfun$fileType$1(FtpClient.scala:304)
	at oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.retry(FtpClient.scala:281)
	at oharastream.ohara.client.filesystem.ftp.FtpClient$Builder$$anon$1.fileType(FtpClient.scala:304)
	at oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.$anonfun$fileType$1(FtpFileSystem.scala:281)
	at oharastream.ohara.client.filesystem.FileSystem.wrap(FileSystem.scala:92)
	at oharastream.ohara.client.filesystem.FileSystem.wrap$(FileSystem.scala:90)
	at oharastream.ohara.client.filesystem.ftp.FtpFileSystem$Builder$FtpFileSystemImpl.wrap(FtpFileSystem.scala:136)
	at 
:     :     :     :
```
==> 資料夾不存在但這裡拋 NoSuchElementException 似乎有點奇怪?

2. 
![screenshot 50](https://user-images.githubusercontent.com/475186/75858527-60499700-5e33-11ea-865e-59a87d69e30a.jpg)
==> 錯誤時無法再 Edit, 導致無法做修改

**Comment by vitojeng** [comment]
metrics 的數字與單位, 距離似乎太遙遠?
![screenshot 52](https://user-images.githubusercontent.com/475186/75940655-a43aab80-5ec8-11ea-9f07-19069ee020d0.jpg)

**Comment by vitojeng** [comment]
在 Property view -> Metrics 打開情形下, 若關閉 Property view 後再度打開, metrics 會處於關閉狀態.
希望能記住之前是否打開, 下次 Property view 打開時可以恢復之前的狀態

**Comment by vitojeng** [comment]
metrics 目前是反應後端 runtime 下的數字, 因此 pipeline component 在 stop 的狀態下, metrics 是否要歸 0 比較合理?

**Comment by vitojeng** [comment]
Internal Topic 的錯誤, 除了 name 之外, 可能還是附上 pipeline 上的名稱, 會比較清楚~
Ex: `Failed to stop topic 59sn5fcc4y(T1)`

![screenshot 53](https://user-images.githubusercontent.com/475186/75941995-2ed0da00-5ecc-11ea-85ce-a4386ae22810.jpg)

**Comment by wu87988622** [comment]
> metrics 目前是反應後端 runtime 下的數字, 因此 pipeline component 在 stop 的狀態下, metrics 是否要歸 0 比較合理?

目前會做清除，但是要等到下一次更新，大約是等待五秒後，就會清掉了

**Comment by konekoya** [comment]
> Internal Topic 的錯誤, 除了 name 之外, 可能還是附上 pipeline 上的名稱, 會比較清楚~
Ex: Failed to stop topic 59sn5fcc4y(T1)

@vitojeng Do you mean it should display __T1__ instead of __59sn5fcc4y__? The latter one actually doesn't mean anything to our users, so we always hide it from the UI :)

**Comment by konekoya** [comment]
> metrics 的數字與單位, 距離似乎太遙遠?

I think it's okay if we put them together anyway. Like `5,998 messages` instead of breaking them apart

Ping @eechih for idea :)

**Comment by konekoya** [comment]
> 在 Property view -> Metrics 打開情形下, 若關閉 Property view 後再度打開, metrics 會處於關閉狀態.
希望能記住之前是否打開, 下次 Property view 打開時可以恢復之前的狀態

**Comment by chia7712** [comment]
> 資料夾不存在但拋 NoSuchElementException 似乎有點奇怪?

Or IllegalArgumentException?

**Comment by eechih** [comment]
> metrics 的數字與單位, 距離似乎太遙遠?

> Internal Topic 的錯誤, 除了 name 之外, 可能還是附上 pipeline 上的名稱, 會比較清楚~
Ex: Failed to stop topic 59sn5fcc4y(T1)

@vitojeng @konekoya 這兩個問題不影響使用者操作，等到下個版本再來討論，OK?

**Comment by vitojeng** [comment]
> @vitojeng @konekoya 這兩個問題不影響使用者操作，等到下個版本再來討論，OK?

OK~ 有需要 backport 的我會再額外提醒

**Comment by vitojeng** [comment]
> > 資料夾不存在但拋 NoSuchElementException 似乎有點奇怪?
> 
> Or IllegalArgumentException?

比 NoSuchElementException 好 10 倍

**Comment by chia7712** [comment]
![註解 2020-03-05 141900](https://user-images.githubusercontent.com/6234750/75956061-14115c00-5ef2-11ea-8217-d9ea367359a9.jpg)

死掉的節點不應該讓使用者能選擇

**Comment by konekoya** [comment]
> @vitojeng @konekoya 這兩個問題不影響使用者操作，等到下個版本再來討論，OK?

No problem :)

**Comment by vitojeng** [comment]
#### 用 FtpSourceConnector 測試大型 csv file
環境: 3 nodes; CPU 2 core, RAM 8 GB each node

0. 建立一個 workspace, 3 nodes.
0. 建立一個 pipeline, 並拉入: ftp source connector --> internal topic --> ftp sink connector
0. Start source and sink connector
0. 放一個 csv 小檔案到 input folder, 確認處理正常: 
    * input folder 移到 completed folder
    * sink connector 有寫入 file. 
    * source/sink component 的 Metrics 顯示正確
0. 再放一個 csv 小檔案到 input folder, 確認處理正常. 同 step 3.
0. 放入一個 csv 大檔案(600MB+)到 input folder. 觀察情形:
    * Metrics: 前數分鐘, FtpSourceConnector 的 Property view 有顯示 Metrics, 但數字沒有更被新. 過更久的時間後, Metrics 即不再顯示(component 顯示 **No metrics data available** )
    * input folder 的 csv 檔案仍然存在
    * 用 Developer Tools/LOGS 觀察三個 node 的 worker, 沒有看到不正常的 exception.
0. 此時再放一個 csv 小檔案, 發現也不會被處理了.

**Comment by vitojeng** [comment]
#### 切換 pipeline 時出現 error: Failed to stop topic XXXXX

* pipeline 1: 有一個 internal topic
* pipeline 2: 完全空的

在 pipeline1 切換到 pipeline 2 時, 出現 error: `Failed to stop topic XXXXX`
而pipeline2 切到 pipeline 1 時, 沒有出現 error.
不知為何在切換 pipeline 時會需要做這個動作?

![screenshot 55](https://user-images.githubusercontent.com/475186/75961440-fe556400-5efc-11ea-9fec-b00364d039e8.jpg)

Detail
![screenshot 56](https://user-images.githubusercontent.com/475186/75961585-41afd280-5efd-11ea-8e01-686ac1a4816b.jpg)

**Comment by konekoya** [comment]
> 確認後若不是太困難, 希望這個 issue 可以 backport 到 0.9.x :)

Vito, there's already a PR addresses this issue. I'll try to get it in today and ping you again for testing. (Sorry, it seems like GitHub's mention feature is broken so I somehow couldn't tag you in this comment)

**Comment by vitojeng** [comment]
related with https://github.com/oharastream/ohara/issues/3706#issuecomment-595061071
dead node 的資訊顯示怪怪的~

![screenshot 58](https://user-images.githubusercontent.com/475186/76043979-f5fc3800-5f93-11ea-88f6-006033309b5e.jpg)
![screenshot 60](https://user-images.githubusercontent.com/475186/76044063-2ba12100-5f94-11ea-82a7-559559149a7c.jpg)

**Comment by vitojeng** [comment]
觀察 Snackbar 目前容易被忽略的因素: 一個是顯示在左下角, 另一個是顯示時間不長.
目前 layout 大部份操作都集中在畫面上半部, 使用者目光停留在上半部機會很高. 在高解晰度的大螢幕下, 左下角的 Snackbar 相信被忽略的機會大增. 因此建議 Snackbar 顯示在畫面中上方, 應該可以增加被使用者注意到的機會~ (希望可以 backport 到 0.9.x)

**Comment by vitojeng** [comment]
#### Property view 會自動關閉

在 Property view 開啟的情形下, 從 Developer Tools/LOGS 切換到 Developer Tools/Topic 時, Property view 會自動被關閉.

**Comment by konekoya** [comment]
> 確認後若不是太困難, 希望這個 issue 可以 backport 到 0.9.x :)

@vitojeng The PR has been merged. But It still needs some time to build the image :)

**Comment by konekoya** [comment]
> 觀察 Snackbar 目前容易被忽略的因素: 一個是顯示在左下角, 另一個是顯示時間不長.
目前 layout 大部份操作都集中在畫面上半部, 使用者目光停留在上半部機會很高. 在高解晰度的大螢幕下, 左下角的 Snackbar 相信被忽略的機會大增. 因此建議 Snackbar 顯示在畫面中上方, 應該可以增加被使用者注意到的機會~ (希望可以 backport 到 0.9.x)

Pinging @oharastream/frontend for thoughts

**Comment by wu87988622** [comment]
> related with [#3706 (comment)](https://github.com/oharastream/ohara/issues/3706#issuecomment-595061071)
> dead node 的資訊顯示怪怪的~
> 
> ![screenshot 61](https://user-images.githubusercontent.com/475186/76044158-70c55300-5f94-11ea-89cb-056cdc9d5de8.jpg)
> 
> ![screenshot 60](https://user-images.githubusercontent.com/475186/76044063-2ba12100-5f94-11ea-82a7-559559149a7c.jpg)

@vitojeng can you get me unavailable node response, i can't reproduce that, thanks

**Comment by vitojeng** [comment]
> @vitojeng can you get me unavailable node response, i can't reproduce that, thanks

Just shutdown the one of kubernetes slave node and you should be able to reproduce it.

**Comment by chia7712** [comment]
![註解 2020-03-09 143009](https://user-images.githubusercontent.com/6234750/76188233-91024580-6212-11ea-9b93-92b30ecd40a5.jpg)

toolbox被log畫面擋住，沒有卷軸可以使用

**Comment by chia7712** [comment]
log畫面開著的時候，pipeline上的"變更"都會觸發重拉log...這是預期的行為嗎?

**Comment by chia7712** [comment]
![註解 2020-03-09 143009](https://user-images.githubusercontent.com/6234750/76188812-0f131c00-6214-11ea-9459-d113ee1f529d.jpg)

同一個workspace，不同pipeline裡面新增一個重複名稱的物件，會顯示錯誤但該物件依然留在畫布上

**Comment by saivirtue** [comment]
> log畫面開著的時候，pipeline上的"變更"都會觸發重拉log...這是預期的行為嗎?

DevTool工具打開的時候，Pipeline上選中的component會觸發搜尋對應的log / topicData:
**DevTool - topic tab :** 
- 選中topic => query topic data
- 選中stream => switch to log tab, query stream log
- 選中connector => switch to log tab, query worker log
**DevTool - log tab :** 
- 選中topic => query broker log
- 選中stream => query stream log
- 選中connector => query worker log

**Comment by chia7712** [comment]
@saivirtue 了解，就是之前討論的選種哪個物件就找尋可能的logs

**Comment by saivirtue** [comment]
> @saivirtue 了解，就是之前討論的選種哪個物件就找尋可能的logs

You are right :)

**Comment by chia7712** [comment]
![註解 2020-03-09 143009](https://user-images.githubusercontent.com/6234750/76189118-d6277700-6214-11ea-892f-c2a32a60ac82.jpg)

如果現在UI嚴格限制只能1-to-1的話，那在連線後就應該把"連線"的按鈕遮掉...

**Comment by konekoya** [comment]
> 同一個workspace，不同pipeline裡面新增一個重複名稱的物件，會顯示錯誤但該物件依然留在畫布上

@chia7712 Hmm, I'm not able to repro this. Not sure if I understand this right:

1. pipeline1 -> create FTP source named "aaa"
2. pipeline2 => create FTP source named "aaa"

This works okay for me

**Comment by chia7712** [comment]
@konekoya it seems a temporary connection error so I remove this comment :)

**Comment by vitojeng** [comment]
在 Create Pipeline 時, 若使用重覆的 name, 會在左下角顯示 "Failed to create pipeline XXX".
比較好的做法應該是 Create Pipeline Dialog 時通知使用者修正. (0.10 only)

**Comment by vitojeng** [comment]
在 Edit Property Dialog, 我們使用 `*` 顯示 password; 但在 Property View 裡並沒有使用 `*`:
![screenshot](https://user-images.githubusercontent.com/475186/76195372-1a217880-6223-11ea-95d1-4fc5c80519c5.jpg)

Please backport to 0.9.x

**Comment by vitojeng** [comment]
目前 Metrics 開關, 似乎是所有 Pipeline 都共用同一組.
每個 Pipeline , 應該自己的 Metrics 開關, 預設是關.

**Comment by vitojeng** [comment]
討論: Pipeline componet name 目前只能小寫與數字. 
使用者在想要以多個單字組合來命名時, 全部小寫會難以辦認, 可能需要討論如何修正.(0.10 only)

**Comment by chia7712** [comment]
![註解 2020-03-10 125154](https://user-images.githubusercontent.com/6234750/76286936-451cd280-62de-11ea-8ba6-21378a685804.jpg)

(我好像之前回報過) unavailable nodes should NOT be selectable!

**Comment by chia7712** [comment]
> 討論: Pipeline componet name 目前只能小寫與數字.

因為k8s對名稱有強烈要求，為了一致性所以讓所有物件都套用相同限制

**Comment by chia7712** [comment]
![註解 2020-03-10 125154](https://user-images.githubusercontent.com/6234750/76287188-cecca000-62de-11ea-8a83-059cd2e16257.jpg)

確認一下，錯誤訊息跑到右上角是預期的行為嗎?

**Comment by konekoya** [comment]
> 確認一下，錯誤訊息跑到右上角是預期的行為嗎?

Yep. This was recently changed in  #4268 :)

**Comment by chia7712** [comment]
![註解 2020-03-10 125154](https://user-images.githubusercontent.com/6234750/76288883-d8580700-62e2-11ea-847b-8a7a23874e09.jpg)

欄位"c"跑版了

**Comment by saivirtue** [comment]
> ![註解 2020-03-10 125154](https://user-images.githubusercontent.com/6234750/76288883-d8580700-62e2-11ea-847b-8a7a23874e09.jpg)
> 
> 欄位"c"跑版了

IIRC, the "latest column" will align field to right.
/cc @konekoya Am I right?

**Comment by chia7712** [comment]
![註解 2020-03-10 125154](https://user-images.githubusercontent.com/6234750/76290286-ec513800-62e5-11ea-9ad1-2cba15313fda.jpg)

應該要禁止加入重複的節點

**Comment by vitojeng** [comment]
#### FtpSource Connector 沒有顯示 Nodes 資訊

* 1 個 workspace, 只有一個 pipeline.
* FtpSource Connector(A) --> Internal topic(B) --> ConsoleSink(C)
* 填入參數, 啟動成功. 等待一會兒, 出現 Metrics Panel
* 上傳 csv file, 確認 A,C 的 metrics 有正確顯示
* 發現 C 有 Nodes 資訊, 但 A 沒有.
* 觀察 Response, API(/connectors) 裡 A 的 taskStatus 是空的, C 的 taskStatus 有值

**Comment by chia7712** [comment]
> 觀察 Response, API(/connectors) 裡 A 的 taskStatus 是空的, C 的 taskStatus 有值

connector.state有值嗎? 這可能是開connector成功，但開task的時候失敗。可以看一下worker logs

**Comment by konekoya** [comment]
> IIRC, the "latest column" will align field to right.
/cc @konekoya Am I right?

Right, but the cell data should also align to the right. I can submit a PR to fix this if you're busy with the API refactoring :)

**Comment by vitojeng** [comment]
Node unavailable 時:
* NaN 應該顯示 `-`
* Status 按照其他的模式, 顯示 `Alive`(綠色) or `Dead`(紅色)

![screenshot 4](https://user-images.githubusercontent.com/475186/76378349-30951480-6388-11ea-84b6-ea7251a20fa9.jpg)

**Comment by konekoya** [comment]
> (我好像之前回報過) unavailable nodes should NOT be selectable!

@chia7712 Thanks for reporting, I will look into it today :)

**Comment by konekoya** [comment]
> CPU/Memory: NaN 應該顯示 -

@vitojeng This will be fixed in #4313 :)

**Comment by vitojeng** [comment]
前幾天有遇到, 現在又遇到...但重現不出來 QQ
![screenshot 6](https://user-images.githubusercontent.com/475186/76392318-43b9db80-63ac-11ea-902f-df58736c9dd0.jpg)

**Comment by konekoya** [comment]
> 更正，該節點並沒有成功被加入到後端，不過畫面上卻有顯示...

補充，嘗試刪除"多出來的節點":

刪除本尊 -> 成功
刪除複製人 -> 失敗，因為刪除的按鈕不能按

@chia7712 I'm unable to repro this. Adding two same name nodes is blocked by the APIs

**Comment by chia7712** [comment]
> I'm unable to repro this. Adding two same name nodes is blocked by the APIs

I will report this issue if I find a way to reproduce this bug :)

**Comment by chia7712** [comment]
> Node unavailable 時:
> CPU/Memory: NaN 應該顯示 -
> State: 應與其他地方一致, 顯示 Alive(綠色) or Dead(紅色)

@oharastream/frontend Is there a issue ?

**Comment by konekoya** [comment]
> CPU/Memory: NaN 應該顯示

Fixed in #4313 :)

> State: 應與其他地方一致, 顯示 Alive(綠色) or Dead(紅色)

Tracked in #4368

**Comment by chia7712** [comment]
all issues are resolved so close this issue now.

feel free to raise issue on #3019 

FYI @oharastream/developer
